source/TimePicker.js

```diff
@@ -585,14 +608,27 @@
 		meridiemPickerChanged: function (sender, event) {
 			if(this.syncingPickers) return true;
 
-			var hour = this.$.hour.get('value'),
-				value = event.value;
+			var meridiems = this.meridiems,
+				hour = this.$.hour.getValue(),
+				oldIndex = event.old,
+				newIndex = this.$.meridiem.getValue(),
+				offset, start, end, newHour;
 
-			if (this.value) {
-				// value is 0 for am, 1 for pm
-				// reset the hour to < 12 and then add 12 if it's pm
-				hour = hour%12 + value*12;
-				this.updateHours(hour);
+			offset = hour - parseInt(meridiems[oldIndex]['start'], 10);
```

Comment:
When we have just two locale like AM/PM, hour and minute are not changed when we change meridiem betw'n AM/PM.
For example, AM 11:45 will become PM 11:45.
I though this behavior is general in other locale too.

Yes, we'd better get confrim from @ehoogerbeets